In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/adult-dataset/adult.data.csv


In [2]:
df = pd.read_csv('/kaggle/input/adult-dataset/adult.data.csv') #reading the dataset, it was uploaded in Kaggle Datasets

In [3]:
df #dataframe

age         workclass   education     marrital status  \
0       39         State-gov   Bachelors       Never-married   
1       50  Self-emp-not-inc   Bachelors  Married-civ-spouse   
2       38           Private     HS-grad            Divorced   
3       53           Private        11th  Married-civ-spouse   
4       28           Private   Bachelors  Married-civ-spouse   
...    ...               ...         ...                 ...   
32556   27           Private  Assoc-acdm  Married-civ-spouse   
32557   40           Private     HS-grad  Married-civ-spouse   
32558   58           Private     HS-grad             Widowed   
32559   22           Private     HS-grad       Never-married   
32560   52      Self-emp-inc     HS-grad  Married-civ-spouse   

              occupation     sex  capital gain  capital loss  \
0           Adm-clerical    Male          2174             0   
1        Exec-managerial    Male             0             0   
2      Handlers-cleaners    Male             0             0   
3      Handlers-cleaners    Male             0             0   
4         Prof-specialty  Female             0             0   
...                  ...     ...           ...           ...   
32556       Tech-support  Female             0             0   
32557  Machine-op-inspct    Male             0             0   
32558       Adm-clerical  Female             0             0   
32559       Adm-clerical    Male             0             0   
32560    Exec-managerial  Female         15024             0   

       working hours per week  salary  
0                          40   <=50K  
1                          13   <=50K  
2                          40   <=50K  
3                          40   <=50K  
4                          40   <=50K  
...                       ...     ...  
32556                      38   <=50K  
32557                      40    >50K  
32558                      40   <=50K  
32559                      20   <=50K  
32560                      40    >50K  

[32561 rows x 10 columns]

In [4]:
df.head() #head of the dataframe

age         workclass  education     marrital status         occupation  \
0   39         State-gov  Bachelors       Never-married       Adm-clerical   
1   50  Self-emp-not-inc  Bachelors  Married-civ-spouse    Exec-managerial   
2   38           Private    HS-grad            Divorced  Handlers-cleaners   
3   53           Private       11th  Married-civ-spouse  Handlers-cleaners   
4   28           Private  Bachelors  Married-civ-spouse     Prof-specialty   

      sex  capital gain  capital loss  working hours per week  salary  
0    Male          2174             0                      40   <=50K  
1    Male             0             0                      13   <=50K  
2    Male             0             0                      40   <=50K  
3    Male             0             0                      40   <=50K  
4  Female             0             0                      40   <=50K

>  ***Data Dictionary***
> * Age: This represents the of the working professional. 
> * Workclass: This represents the class of work they are
> * Education: This represents the level of education.
> * Marrital Status: Represnts the marital status of an individual.
> * Occupation: The kind of occupation an adult is involved in. 
> * Sex: Gender of the Adult
> * Capital Gain: The gain they have made out of their capital Investment. 
> * Capital Loss: The loss they have incurred out if their capital Investment. 
> * working hours per week: Number of hours an individual has worked in a week. 
> * Salary (Target): Represents of the salary is less the or greater than 50K

In [10]:
df.info() #information about the dataset

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32561 entries, 0 to 32560
Data columns (total 10 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   age                     32561 non-null  int64 
 1   workclass               32561 non-null  object
 2   education               32561 non-null  object
 3   marrital status         32561 non-null  object
 4   occupation              32561 non-null  object
 5   sex                     32561 non-null  object
 6   capital gain            32561 non-null  int64 
 7   capital loss            32561 non-null  int64 
 8   working hours per week  32561 non-null  int64 
 9   salary                  32561 non-null  object
dtypes: int64(4), object(6)
memory usage: 2.5+ MB


> The above information show that there are 32561 entries in the dataset with 4 integer type columns and 6 object type columns, and there re no missing values as the Non-Null Count for all the columns is equasl to the number of entries. 

In [15]:
df.isnull().sum() #Checking for any missing values. 

age                       0
workclass                 0
education                 0
marrital status           0
occupation                0
sex                       0
capital gain              0
capital loss              0
working hours per week    0
salary                    0
dtype: int64

> We will check each column for its quality. 

In [23]:
for i in df.columns:
    print(i.capitalize())
    print(df[i].unique())
    print('- ' * 50)

Age
[39 50 38 53 28 37 49 52 31 42 30 23 32 40 34 25 43 54 35 59 56 19 20 45
 22 48 21 24 57 44 41 29 18 47 46 36 79 27 67 33 76 17 55 61 70 64 71 68
 66 51 58 26 60 90 75 65 77 62 63 80 72 74 69 73 81 78 88 82 83 84 85 86
 87]
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
Workclass
['State-gov' 'Self-emp-not-inc' 'Private' 'Federal-gov' 'Local-gov' '?'
 'Self-emp-inc' 'Without-pay' 'Never-worked']
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
Education
['Bachelors' 'HS-grad' '11th' 'Masters' '9th' 'Some-college' 'Assoc-acdm'
 'Assoc-voc' '7th-8th' 'Doctorate' 'Prof-school' '5th-6th' '10th'
 '1st-4th' 'Preschool' '12th']
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
Marrital status
['Never-married' 'Married-civ-spouse' 'Divorced' 'Married-spouse-absent'
 'Separated' 'Married-AF-spouse' 'Widowed']
- - - - - - - - - - - - - -

> We can see age has a value 17, which can be invalid as it is not a leagal age to work.


37.49298006528698